back/tests/WebSpec.hs

```diff
@@ -25,23 +26,92 @@ import Control.Monad.Catch
 -- Tests
 -----------------------------------------------------------------------------
 
+logFile :: FilePath
+logFile = "/tmp/test_guide.log"
+
 tests :: IO ()
-tests = run $ do
-  mainPageTests
-  categoryTests
-  itemTests
-  markdownTests
+tests = do
+  run $ do
+    mainPageTests
+    categoryTests
+    itemTests
+    markdownTests
+    Api.tests
+  hspec logTest
+
+getLines :: Handle -> IO String
+getLines h = loop' []
+  where
+    loop' :: [String] -> IO String
+    loop' xs = do
+      eLine <- try $ hGetLine h
+      case eLine of
+        Left (_ :: SomeException) -> pure $ concat $ reverse xs
+        Right line                -> loop' (line:xs)
+
+logTest :: Spec
+logTest = H.describe "test of logger" $ do
+  logs <- H.runIO $ do
+    logFileHandle <- openFile logFile ReadWriteMode
+    logs <- getLines logFileHandle
+    hClose logFileHandle
+    writeFile logFile ""
+    pure logs
+  H.describe "Logging of init" $ do
+    H.it "spock is runned" $ [re|Spock is running on port|] `isIn` logs
+    H.it "api is runned" $ [re|API is running on port|] `isIn` logs
+  H.describe "Logging of api" $ do
+    H.describe "Categories" $ do
+      H.it "get categories request" $ [re|getCategories|] `isIn` logs
+      H.it "create ctegory request" $ [re|createCategory|] `isIn` logs
+      H.it "get category by id" $ [re|getCategory|] `isIn` logs
+      H.it "delete category by id request" $ [re|deleteCategory|] `isIn` logs
+      H.it "modify notes to category request" $ [re|setCategoryNotes|] `isIn` logs
+      H.it "modify info to category request" $ [re|setCategoryInfo|] `isIn` logs
+    H.describe "Item" $ do
+      H.it "create item" $ [re|createItem|] `isIn` logs
+      H.it "delete item by id" $ [re|deleteItem|] `isIn` logs
+      H.it "get item by id" $ [re|getItem|] `isIn` logs
+      H.it "set item info" $ [re|setItemInfo|] `isIn` logs
+      H.it "set item summary" $ [re|setItemSummary|] `isIn` logs
+      H.it "set item ecosystem" $ [re|setItemEcosystem|] `isIn` logs
+      H.it "set item notes" $ [re|setItemNotes|] `isIn` logs
+    H.describe "Trait" $ do
+      H.it "create new trait" $  [re|createTrait|] `isIn` logs
+      H.it "delete trait" $ [re|deleteTrait|] `isIn` logs
+      H.it "get trait" $ [re|getTrait|] `isIn` logs
+      H.it "modify trait" $ [re|setTrait|] `isIn` logs
+      H.it "move trait" $ [re|moveTrait|] `isIn` logs
+  
+    H.describe "Errors (exceptions)" $ do
+      H.it "Category not found" $
+        [re|ServantErr {errHTTPCode = 404, errReasonPhrase = "Category not found"|] `isIn` logs
+      H.it "Merge conflict occurred" $
+        [re|ServantErr {errHTTPCode = 409, errReasonPhrase = "Merge conflict occurred"|] `isIn` logs
+      H.it "Item not found" $
+        [re|ServantErr {errHTTPCode = 404, errReasonPhrase = "Item not found"|] `isIn` logs
+
+
+
+isIn :: RE -> String -> H.Expectation
+isIn  reg text = (`H.shouldBe` True) $ matched $ text ?=~ reg
```

Comment:
It's better to write something with `H.expectationFailure` because then you'll get the actual regex that failed, and not "expected True, got False".

You should also add `HasCallStack =>` to get stack traces.